integration-tests/src/test/groovy/com/okta/sdk/tests/it/EventHooksIT.groovy

```diff
@@ -0,0 +1,254 @@
+                .setItems(["user.lifecycle.create", "user.lifecycle.activate"]))
+            .setChannel(client.instantiate(EventHookChannel)
+                .setType(EventHookChannel.TypeEnum.HTTP)
+                .setVersion("1.0.0")
+                .setConfig(client.instantiate(EventHookChannelConfig)
+                    .setUri("https://www.example.com/eventHooks")
+                    .setHeaders([ client.instantiate(EventHookChannelConfigHeader)
+                                      .setKey("X-Test-Header")
+                                      .setValue("Test header value")])
+                    .setAuthScheme(client.instantiate(EventHookChannelConfigAuthScheme)
+                        .setType(EventHookChannelConfigAuthSchemeType.HEADER)
+                        .setKey("Authorization")
+                        .setValue("Test-Api-Key"))))
+        registerForCleanup(createdEventHook)
+
+        assertThat(createdEventHook.getId(), notNullValue())
+
+        EventHook retrievedEventHook = client.getEventHook(createdEventHook.getId())
+        assertThat(retrievedEventHook.getId(), equalTo(createdEventHook.getId()))
+
+        createdEventHook.deactivate()
+        createdEventHook.delete()
+
+        sleep(2000)
+
+        try {
+            client.getEventHook(createdEventHook.getId())
+        }
+        catch (ResourceException e) {
+            assertThat(e.status, equalTo(404))
+        }
+    }
+
+    @Test
+    void listAllEventHooksTest() {
+        String name = "event-hook-java-sdk-${UUID.randomUUID().toString()}"
+
+        EventHookList eventHookList = client.listEventHooks()
+
+        int listSize = eventHookList.size()
+
+        EventHook createdEventHook = client.instantiate(EventHook)
+            .setName(name)
+            .setEvents(client.instantiate(EventSubscriptions)
+                .setType(EventSubscriptions.TypeEnum.EVENT_TYPE)
+                .setItems(["user.lifecycle.create", "user.lifecycle.activate"]))
+            .setChannel(client.instantiate(EventHookChannel)
+                .setType(EventHookChannel.TypeEnum.HTTP)
+                .setVersion("1.0.0")
+                .setConfig(client.instantiate(EventHookChannelConfig)
+                    .setUri("https://www.example.com/eventHooks")
+                    .setHeaders([ client.instantiate(EventHookChannelConfigHeader)
+                                      .setKey("X-Test-Header")
+                                      .setValue("Test header value")])
+                    .setAuthScheme(client.instantiate(EventHookChannelConfigAuthScheme)
+                        .setType(EventHookChannelConfigAuthSchemeType.HEADER)
+                        .setKey("Authorization")
+                        .setValue("Test-Api-Key"))))
+        registerForCleanup(createdEventHook)
+
+        assertThat(createdEventHook.getId(), notNullValue())
+
+        sleep(2000)
+
+        assertThat(client.listEventHooks(), hasSize(listSize + 1))
```

Comment:
This is prone to error if anything else changes in the org related to hooks.  Your best bet it just check if the hook has been added to the list, something like:
``` groovy
client.listEventHooks().anyMatch { createdEventHook.getId().equals(it.id) }
```

Reply:
Noted.